### DISCUSSION TITLE
Data Links: Add panel's individual "from" and "to" timestmap variables

### DISCUSSION THREAD

**Discussion by clompsy** [grafana/grafana#70363] [Dashboards, Panels & Visualizations]
Currently when adding a data link to a panel, we have the built-in variables ${__from} and ${__to}, those reflect the dashboard's timerange as intended.
What we're missing are built-in variables to get the panel's individual from and to timestamps if it has it's own "relative time" or "timeshift" parameters to be able to use links with the panel's specific timerange.
Most datasource plugins have their own variables to cover this, but the data links don't.

As of now there is no way to get the from/to for the panel's individual timerange which makes usage of data links in such cases quite complicated.

Suggestion: Add the variables ${__panelFrom} and ${__panelTo} variables as "Built-In" to be used in data links.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82652. Feel free to continue the discussion around this there. Thank you!
